### PR TITLE
Fixed `odoo.dob.field` and `odoo.weight.field` properties and 'patientVoided' → 'voided'.

### PR DESCRIPTION
**Pull request by icrc-loliveira**
Fixed property access for:
- odoo.dob.field
- odoo.weight.field

Fixed property name for patient voided.

### CODE REVIEW

**Comment by icrc-loliveira** [comment]
cc: @icrc-fdeniger

File: src/main/resources/camel/odoo-patient-handler.xml
Comment (wluyima) [inline src/main/resources/camel/odoo-patient-handler.xml:26]: I think you might want to check both `voided` and `patientVoided`, a person can be linked to more than one entity e.g. User, Provider and Patient. Therefore, a patient account can get voided but not necessarily their person record, or user or provider accounts.

File: src/main/resources/camel/odoo-add-extra-quotation-details.xml
Comment (wluyima) [inline src/main/resources/camel/odoo-add-extra-quotation-details.xml:13]: Is this meant to address a bug?

File: src/main/resources/camel/odoo-add-extra-customer-details.xml
Comment (wluyima) [inline src/main/resources/camel/odoo-add-extra-customer-details.xml:13]: Is this meant to address a bug?

File: src/main/resources/camel/odoo-add-extra-quotation-details.xml
Comment (wluyima) [inline src/main/resources/camel/odoo-add-extra-quotation-details.xml:16]: Is this meant to address a bug?

**Review by wluyima — COMMENTED** [review]
I added some comments

File: src/main/resources/camel/odoo-patient-handler.xml
Comment (VaishSiddharth) [reply src/main/resources/camel/odoo-patient-handler.xml:26]: @wluyima it seems like we have `voided` attribute in `patient` model but AFAIK we don't have any attribute as `patientVoided`
We have `personVoided` in `person` model.
CMIIW

Comment (icrc-loliveira) [reply src/main/resources/camel/odoo-patient-handler.xml:26]: Hello @wluyima,
I am not complete sure how this data structure works, but here it seems that it's getting the `patient` info and not the `person`, so that should not happen. Or are you saying this `patient` is actually a `person` but if that's the case it should not know if the `patient` is voided or not, since as you pointed it may not be a `patient` at all.

File: src/main/resources/camel/odoo-add-extra-customer-details.xml
Comment (icrc-loliveira) [reply src/main/resources/camel/odoo-add-extra-customer-details.xml:13]: Yes.

File: src/main/resources/camel/odoo-add-extra-quotation-details.xml
Comment (icrc-loliveira) [reply src/main/resources/camel/odoo-add-extra-quotation-details.xml:13]: Yes.

File: src/main/resources/camel/odoo-add-extra-quotation-details.xml
Comment (icrc-loliveira) [reply src/main/resources/camel/odoo-add-extra-quotation-details.xml:16]: Yes.

File: src/main/resources/camel/odoo-add-extra-customer-details.xml
Comment (Ruhanga) [reply src/main/resources/camel/odoo-add-extra-customer-details.xml:13]: I think this is actually introducing a bug... @icrc-loliveira have you tested this? Or did you intend to do as below?
```suggestion
                    <simple>${properties:odoo.dob.field}</simple>
```
Which kind of bug is being fixed here though?

File: src/main/resources/camel/odoo-add-extra-quotation-details.xml
Comment (Ruhanga) [reply src/main/resources/camel/odoo-add-extra-quotation-details.xml:13]: See my comment above...

**Review by Ruhanga — COMMENTED** [review]
Hi @icrc-loliveira, thanks for the PR. I just have a few concerns below.

**Comment by icrc-loliveira** [comment]
Hello @Ruhanga and @wluyima,
I have done the requested changes, can you please finalize the review?

**Comment by delphinepas** [comment]
@wluyima @Ruhanga could you please review the changes done by  @icrc-loliveira and confirm if it's ok now? thank you cc @icrc-fdeniger

File: src/main/resources/camel/odoo-patient-handler.xml
Comment (wluyima) [reply src/main/resources/camel/odoo-patient-handler.xml:26]: Sorry! That's what I meant to check both Person.personVoided and Patient.voided

**Review by wluyima — COMMENTED** [review]
Responded to comment

File: src/main/resources/camel/odoo-patient-handler.xml
Comment (icrc-fdeniger) [reply src/main/resources/camel/odoo-patient-handler.xml:26]: @VaishSiddharth  @wluyima  why doing test on both properties personVoided and Patient.voided  as in this route we deal with patient data only ( and with patient identifier,...) and only patient are synchronized with Odoo.


IMHO, The use case "a patient is voided but the person is not voided" should be ignored as well  as this route need patients only. If a paitent is voided, we will have an error when trying to get the patient identifier and it causes erros in this route when a patient is voided.

Why should the use case "a patient is voided but the person is not voided" be processed  in this route ?

Comment (wluyima20) [reply src/main/resources/camel/odoo-patient-handler.xml:26]: If the route only processes patients, then it's all good.

**Comment by wluyima20** [comment]
Looks good to me

**Comment by icrc-fdeniger** [comment]
@rbuisson  do you believe we can merge so that I can test the integration in our deployments ?

**Comment by icrc-fdeniger** [comment]
@wluyima  I believe you should add your review here ( it seems you have 2 accounts :))

**Review by wluyima — APPROVED** [review]
Looks good

**Review by icrc-fdeniger — APPROVED** [review]

**Comment by icrc-fdeniger** [comment]
Thanks. We still need an approval from a maintainer. @Ruhanga @VaishSiddharth  @wluyima  or @rbuisson is it on you ?

**Review by mks-d — APPROVED** [review]

**Pull request merged**
